site2/docs/concepts-architecture-overview.md

```diff
@@ -29,6 +29,10 @@ Messages are typically dispatched out of a [managed ledger](#managed-ledgers) ca
 
 Finally, to support geo-replication on global topics, the broker manages replicators that tail the entries published in the local region and republish them to the remote region using the Pulsar [Java client library](client-libraries-java.md).
 
+> **Note**
+> 
+> Broker is thread-safe when running on 64-bit JVM.
```

Comment:
I don't think it's necessary to add a note here, the issues that https://github.com/apache/pulsar/pull/10280 mention are not serious problems? it's might affect the Ledger rollover?

@lhotari WDYT?

Reply:
> I don't think it's necessary to add a note here, the issues that #10280 mention are not serious problems? it's might affect the Ledger rollover?

@codelipenghui There are several serious thread-safety problems if Broker would be run on 32-bit JVM. There are multiple locations where it is assumed that writing of long and double fields are atomic. This assumption is valid on 64-bit JVMs, but not on 32-bit JVMs. When reviewing broker code, it's necessary to know that it's fine to assume that long and double field write are atomic since it will be only run on 64-bit JVMs and there isn't a risk of the "word tearing problem" that happens on 32-bit JVMs for long/double field write.
As discussed in #10280, the resolution would be to simply document that Broker is only supported on 64-bit JVMs. It's very unlikely that there would be anyone trying to run it on 32-bit JVMs. I think the impact of this clarification is greater for code reviews where issues such as the one in #10280 aren't reported as issues when everyone knows that it's fine to assume that long and double field writes are atomic (since we only support 64-bit JVMs).

@Anonymitaet I think it might be better to omit "thread-safe" in the sentence. It's probably sufficient to say that Pulsar Broker is only supported on 64-bit JVMs.  This applies only to Pulsar Broker code. I guess that the shared Pulsar common code and Pulsar client is supported on both 32-bit and 64-bit JVMs for Java 8+.
Perhaps this requirement of the 64-bit JVM can be mentioned in the documentation where it has other requirements for the broker? (is there some documentation about minimum memory requirements etc?)

Reply:
@lhotari many thanks for your details!

How about adding the contents in these two places?

- In [`Deployment > Bare metal > Deploy Pulsar brokers`](https://pulsar.apache.org/docs/en/next/deploy-bare-metal/#deploy-pulsar-brokers)
![image](https://user-images.githubusercontent.com/50226895/118612160-15296300-b7f0-11eb-9d91-cb6b51c45791.png)

- In [`Deployment > Bare metal multi-cluster > Deploy Pulsar brokers`](https://pulsar.apache.org/docs/en/next/deploy-bare-metal-multi-cluster/#deploy-brokers)
![image](https://user-images.githubusercontent.com/50226895/118611992-f1661d00-b7ef-11eb-8ef9-cf9bca7d226c.png)

Reply:
I guess that's one option. 

The documentation is a bit fragmented. There's a "System requirements" or "requirements" section in https://github.com/apache/pulsar/blob/master/site2/docs/getting-started-standalone.md#system-requirements , https://github.com/apache/pulsar/blob/master/site2/docs/deploy-bare-metal.md#requirements and https://github.com/apache/pulsar/blob/master/site2/docs/deploy-bare-metal-multi-cluster.md#system-requirement .

Perhaps the 64 bit JVM requirement belongs to the system requirements section?

Reply:
@lhotari yes, 64-bit JVM belongs to the system requirements. 

A quick question: for standalone, bare mental, and bare mental (multi-cluster), `Note: broker is only supported on 64-bit JVM` is suitable for these 3 situations? 

If so, to make the instructions more clear, I think we can add this note to the `standalone` chapter as well.
![image](https://user-images.githubusercontent.com/50226895/118748330-3ccf0880-b88e-11eb-89c1-6128f46b48f1.png)

@codelipenghui may we know your feedback on this issue? Thanks.

Reply:
OK, LGTM